sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -691,7 +692,8 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
   def warehousePath: String = {
-    getConf(WAREHOUSE_PATH).replace("${system:user.dir}", System.getProperty("user.dir"))
+    new Path(getConf(WAREHOUSE_PATH).replace("${system:user.dir}",
+      System.getProperty("user.dir"))).toUri.toString
```

Comment:
As `warehousePath` will be used like `new Path(warehousePath)`, this will introduce a regression when the path contains spaces or other special characters. E.g.,

```
scala> new Path("foo bar")
res4: org.apache.hadoop.fs.Path = foo bar

scala> new Path(new Path("foo bar").toUri.toString)
res5: org.apache.hadoop.fs.Path = foo%20bar
```

@avulanov could you try the following code in Windows and see if it works?
```
new Path(new Path(getConf(WAREHOUSE_PATH).replace("${system:user.dir}",
       System.getProperty("user.dir"))).toUri).toString
```